src/core/models/error-response.ts

```diff
@@ -0,0 +1,40 @@
+import { StatusCodes } from "http-status-codes";
+
+/**
+ * ErrorResponse class for handling error responses.
+ * @usage
+```
+let errorResponse = ErrorResponse.badRequest('Invalid input', ['Name is required']);
+```
+ */
+export class ErrorResponse {
+  responseType: string;
```

Comment:
create an enum for response type with values success and error

Reply:
I think ìs better to use the native typescript lib for response type:

```typescript
type ResponseType = "basic" | "cors" | "default" | "error" | "opaque" | "opaqueredirect";
```